eos-upgrade-eos2-to-eos3

```diff
@@ -0,0 +1,222 @@
+eval set -- "$ARGS"
+
+usage() {
+    cat <<EOF
+Remove all legacy apps and upgrade from eos2 to eos3
+Usage:
+   $0
+Options:
+   -f,--force	don't ask for confirmation
+   -s,--skip    skip the final step to actually perform the ostree upgrade
+   -h,--help	show this message
+EOF
+}
+
+FORCE=false
+SKIP=false
+while true; do
+    case "$1" in
+        -f|--force)
+            FORCE=true
+            shift
+            ;;
+        -s|--skip)
+            SKIP=true
+            exit 0
+            ;;
+        -h|--help)
+            usage
+            exit 0
+            ;;
+        --)
+            shift
+            break
+            ;;
+    esac
+done
+
+USERID=$(id -u)
+if [ "$USERID" != "0" ]; then
+    echo "Program requires superuser privileges"
+    exit 1
+fi
+
+LATEST_VERSION=2.6.10
+version=`grep VERSION= /etc/os-release | awk -F '"' '{print $2}'`
+major_version=`echo ${version} | awk -F '.' '{print $1}'`
+
+if [ ${version} != ${LATEST_VERSION} ]; then
+    echo "This script only works with Endless OS version ${LATEST_VERSION}"
+    if [ ${major_version} == 2 ]; then
+	echo "Please update the OS and try again"
+    elif [ ${major_version} == 3 ]; then
+	echo "You already are running Endless OS version 3"
+    else
+	echo "You are running an unsupported version: ${version}"
+    fi
+    exit 1
+fi
+
+if ! $FORCE; then
+    echo
+    echo "You are about to upgrade Endless OS from version 2 to version 3."
+    echo
+    echo "Endless OS 3 uses a new technology for applications called flatpak,"
+    echo "which is a standard format for deploying Linux applications that enables"
+    echo "Endless to run applications distributed by other development teams."
+    echo
+    echo "The legacy application format is no longer supported, and the first step"
+    echo "of the upgrade process is to remove all installed application bundles."
+    echo "Then, roughly a gigabyte of data will be downloaded from the internet,"
+    echo "which may take a long time and incur data charges depending on your"
+    echo "internet connection and contract with your provider."
+    echo
+    echo "In the meantime, while downloading Endless OS 3, you may continue"
+    echo "to use Endless OS 2 and some of the core applications such as"
+    echo "the internet browser and office applications."
+    echo "Once the upgrade to Endless OS 3 is downloaded and ready, you will be"
+    echo "asked to reboot, and then you will be able to install new flatpak apps"
+    echo "from the Endless OS 3 app center."
+    echo
+    echo "Once the download completes, it will take several additional minutes"
+    echo "for the deployment to complete -- please be patient and wait until"
+    echo "you see the 'Upgrade complete!' message."
+    echo
+    read -p "Are you sure you want to remove all legacy app bundles and proceed? [y/N] "
+    response=${REPLY,,} # to lower
+    if [[ ! $response =~ ^(yes|y)$ ]]; then
+        exit 1
+    fi
+fi
+
+# Stop any updates in progress
```

Comment:
You have to kill ostree-daemon since it's spawned via dbus. However, the names changed about 2.4 timeframe. Here's my process for stopping the automatic updater reliably on 2.3:
```
systemctl stop eos-updater.timer
systemctl stop eos-updater.service
killall ostree-daemon
```
On 2.4+:
```
systemctl stop eos-autoupdater.timer
systemctl stop eos-autoupdater.service
killall eos-updater
```